### PR TITLE
Updated documentation for ml.inference docs.

### PR DESCRIPTION
**Pull request by ryanthompson591**
Added documentation updates, mostly cleaning the documents for formatting but adding some extra clarifications in module documentation.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by ryanthompson591** [comment]
R: @yeandy 
R: @TheNeuralBit 
R: @AnandInguva

File: sdks/python/apache_beam/ml/inference/sklearn_inference.py [hunk 1]
```diff
@@ -76,29 +85,47 @@ def load_model(self) -> BaseEstimator:
...
+    supports pandas dataframes.
+
+    Example Usage:
+      pcol | RunInference(SklearnModelHandlerPandas(model_uri="my_uri"))
+
+    NOTE::
+      This API and its implementation are under development and
+      do not provide backward compatibility guarantees.
+
+
```
Comment (yeandy) [inline sdks/python/apache_beam/ml/inference/sklearn_inference.py:123]: nit: extra line?

File: sdks/python/apache_beam/ml/inference/base.py [hunk 2]
```diff
@@ -19,15 +19,12 @@
 
 """An extensible run inference transform.
 
-Users of this module can extend the ModelHandler class for any MLframework. Then
-pass their extended ModelHandler object into RunInference to create a
-RunInference Beam transform for that framework.
+Users of this module can extend the ModelHandler class for any machine learning
+framework. Then pass their extended ModelHandler object into RunInference to
```
Comment (yeandy) [inline sdks/python/apache_beam/ml/inference/base.py:23]: nit: grammar for "Then pass their extended..."

File: sdks/python/apache_beam/ml/inference/pytorch_inference.py [hunk 3]
```diff
@@ -170,6 +183,9 @@ def run_inference(
 
     For the same key across all examples, this will stack all Tensors values
     in a vectorized format to optimize the inference call.
+
```
Comment (yeandy) [inline sdks/python/apache_beam/ml/inference/pytorch_inference.py:186]: Can we add inputs here?

File: sdks/python/apache_beam/ml/inference/sklearn_inference.py [hunk 4]
```diff
@@ -76,29 +85,47 @@ def load_model(self) -> BaseEstimator:
   def run_inference(
       self, batch: Sequence[numpy.ndarray], model: BaseEstimator,
       **kwargs) -> Iterable[PredictionResult]:
+    """Runs inferences on a batch of numpy arrays.
+
```
Comment (yeandy) [inline sdks/python/apache_beam/ml/inference/sklearn_inference.py:89]: Can we add inputs here?

**Review by tvalentyn — COMMENTED** [review]
Thanks for fixing these!

File: sdks/python/apache_beam/ml/inference/pytorch_inference.py [hunk 5]
```diff
@@ -99,6 +102,9 @@ def run_inference(
 
     This method stacks the list of Tensors in a vectorized format to optimize
     the inference call.
+
+    Returns:
+      An Iterable of type PredictionResult.
```
[the remark this answers is not in the document]
Comment (ryanthompson591) [reply sdks/python/apache_beam/ml/inference/pytorch_inference.py:107]: Heejong is making a prediction result change. So I'm going to leave that for now.

File: sdks/python/apache_beam/ml/inference/pytorch_inference.py [hunk 6]
```diff
@@ -55,21 +55,24 @@ def _convert_to_device(examples: torch.Tensor, device) -> torch.Tensor:
...
-    :param state_dict_path: path to the saved dictionary of the model state.
-    :param model_class: class of the Pytorch model that defines the model
-    structure.
-    :param device: the device on which you wish to run the model. If
-    ``device = GPU`` then a GPU device will be used if it is available.
-    Otherwise, it will be CPU.
+    """Implementation of the ModelHandler interface for PyTorch.
+
+    Example Usage:
+      pcol | RunInference(PytorchModelHandlerTensor(state_dict_path="my_uri"))
```
[the remark this answers is not in the document]
Comment (ryanthompson591) [reply sdks/python/apache_beam/ml/inference/pytorch_inference.py:67]: ok will change that here and everywhere else.

File: sdks/python/apache_beam/ml/inference/sklearn_inference.py (hunk 4 above)
Comment (ryanthompson591) [reply sdks/python/apache_beam/ml/inference/sklearn_inference.py:89]: sure.

File: sdks/python/apache_beam/ml/inference/pytorch_inference.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (tvalentyn) [reply sdks/python/apache_beam/ml/inference/pytorch_inference.py:107]: it's already merged.

File: sdks/python/apache_beam/ml/inference/pytorch_inference.py (hunk 3 above)
Comment (ryanthompson591) [reply sdks/python/apache_beam/ml/inference/pytorch_inference.py:186]: Done.  @yeandy I've made these inputs fairly light. I was able to make the sklearn inputs more specific. Is there any specificity you recommend for these parameters?

File: sdks/python/apache_beam/ml/inference/pytorch_inference.py [hunk 7]
```diff
@@ -55,21 +55,24 @@ def _convert_to_device(examples: torch.Tensor, device) -> torch.Tensor:
...
-    :param state_dict_path: path to the saved dictionary of the model state.
-    :param model_class: class of the Pytorch model that defines the model
-    structure.
-    :param device: the device on which you wish to run the model. If
-    ``device = GPU`` then a GPU device will be used if it is available.
-    Otherwise, it will be CPU.
+    """Implementation of the ModelHandler interface for PyTorch.
+
+    Example Usage:
+      pcoll | RunInference(PytorchModelHandlerTensor(state_dict_path="my_uri"))
```
Comment (TheNeuralBit) [inline sdks/python/apache_beam/ml/inference/pytorch_inference.py:67]: nit: I think this needs a newline and an extra colon to render as a code block, see for example https://github.com/apache/beam/blob/master/sdks/python/apache_beam/dataframe/convert.py#L50-L55

```suggestion
    Example Usage::
    
      pcoll | RunInference(PytorchModelHandlerTensor(state_dict_path="my_uri"))
```

**Review by TheNeuralBit — APPROVED** [review]
One minor thing I noticed

**Review by yeandy — APPROVED** [review]

File: sdks/python/apache_beam/ml/inference/pytorch_inference.py [hunk 8]
```diff
@@ -190,6 +194,10 @@ def run_inference(
     For the same key across all examples, this will stack all Tensors values
     in a vectorized format to optimize the inference call.
 
+    Args:
+      batch: A sequence of Tensors.
+      model: A pytorch model.
```
[the remark this answers is not in the document]
Comment (AnandInguva) [reply sdks/python/apache_beam/ml/inference/pytorch_inference.py:199]: Do we need to explicitly say `Must implement forward(X)`?  

Also Users passes X as `dict` when the forward call of the model requires multiple positional arguments. If `X` is a `dict`, we unpack the positional arguments to the forward call in RunInference API by calling `model(**X)` which implicitly means `model.forward(**X)`. I think line 200 would confuse the users on this behavior.

[the remark this answers is not in the document]
Comment (yeandy) [reply sdks/python/apache_beam/ml/inference/pytorch_inference.py:199]: Anand and I review this. @ryanthompson591 Not sure if/when you are reviewing my suggestions, but they have just been updated.

File: sdks/python/apache_beam/ml/inference/pytorch_inference.py [hunk 9]
```diff
@@ -103,6 +106,13 @@ def run_inference(
 
     This method stacks the list of Tensors in a vectorized format to optimize
     the inference call.
+
+    Args:
+      batch: A sequence of Tensors.
+      model: A pytorch model.
```
[the remark this answers is not in the document]
Comment (yeandy) [reply sdks/python/apache_beam/ml/inference/pytorch_inference.py:112]: Anand and I review this. @ryanthompson591 Not sure if/when you are reviewing my suggestions, but they have just been updated.

**Review by yeandy — APPROVED** [review]
LGTM after these two suggestions are made
https://github.com/apache/beam/pull/21868#discussion_r898278520
https://github.com/apache/beam/pull/21868#discussion_r898253546

**Comment by pabloem** [comment]
have Andy's suggestions been addressed?

**Comment by yeandy** [comment]
Great, they've been applied. Should be good once doc tests pass

File: sdks/python/apache_beam/ml/inference/pytorch_inference.py [hunk 10]
```diff
@@ -112,41 +128,48 @@ def run_inference(
...
-    :param state_dict_path: path to the saved dictionary of the model state.
-    :param model_class: class of the Pytorch model that defines the model
-    structure.
-    :param device: the device on which you wish to run the model. If
-    ``device = GPU`` then a GPU device will be used if it is available.
-    Otherwise, it will be CPU.
+    """Implementation of the ModelHandler interface for PyTorch.
+
+    Example Usage:
+      pcoll | RunInference(
```
Comment (yeandy) [inline sdks/python/apache_beam/ml/inference/pytorch_inference.py:157]:
```suggestion
    Example Usage::

      pcoll | RunInference(
```

File: sdks/python/apache_beam/ml/inference/sklearn_inference.py [hunk 11]
```diff
@@ -76,13 +77,21 @@ def _validate_inference_args(inference_args):
...
-  """
   def __init__(
       self,
       model_uri: str,
       model_file_type: ModelFileType = ModelFileType.PICKLE):
+    """ Implementation of the ModelHandler interface for scikit-learn
+    using numpy arrays as input.
+
+    Example Usage:
+      pcoll | RunInference(SklearnModelHandlerNumpy(model_uri="my_uri"))
```
Comment (yeandy) [inline sdks/python/apache_beam/ml/inference/sklearn_inference.py:88]:
```suggestion
    Example Usage::

      pcoll | RunInference(SklearnModelHandlerNumpy(model_uri="my_uri"))
```

Comment (TheNeuralBit) [reply sdks/python/apache_beam/ml/inference/sklearn_inference.py:88]: Let's do these in a follow-up PR

**Comment by TheNeuralBit** [comment]
PreCommit failures are unrelated flakes.

File: sdks/python/apache_beam/ml/inference/pytorch_inference.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (tvalentyn) [reply sdks/python/apache_beam/ml/inference/pytorch_inference.py:121]: Specifically, https://github.com/apache/beam/pull/17773 was the PR and documentation was not included.

File: sdks/python/apache_beam/ml/inference/base.py [hunk 12]
```diff
@@ -91,24 +88,43 @@ def run_inference(
...
       model: ModelT,
       inference_args: Optional[Dict[str, Any]] = None) -> Iterable[PredictionT]:
-    """Runs inferences on a batch of examples and
-    returns an Iterable of Predictions."""
+    """Runs inferences on a batch of examples.
+
+    Args:
+      batch: A sequence of examples or features.
+      model: The model used to make inferences.
+
```
[the remark this answers is not in the document]
Comment (ryanthompson591) [reply sdks/python/apache_beam/ml/inference/base.py:96]: resolved in https://github.com/apache/beam/pull/21921

File: sdks/python/apache_beam/ml/inference/pytorch_inference.py (hunk 7 above)
Comment (ryanthompson591) [reply sdks/python/apache_beam/ml/inference/pytorch_inference.py:68]: resolved in https://github.com/apache/beam/pull/21921

**Pull request merged**